evmap-derive/tests/lib.rs

```diff
@@ -0,0 +1,28 @@
+use evmap_derive::ShallowCopy;
+use std::sync::Arc;
+
+#[derive(ShallowCopy)]
+enum Message {
+	Quit,
+	ChangeColor(i32, i32, i32),
+	Move { x: i32, y: i32 },
+	Write(String)
+}
+
+struct Shallow;
+
+impl evmap::ShallowCopy for Shallow {
+	unsafe fn shallow_copy(&self) ->  std::mem::ManuallyDrop<Self> {
+		unimplemented!();
+	}
+}
+
+#[derive(ShallowCopy)]
```

Comment:
We should probably also have some tests for tuple structs and for generic types. It'd also be good to have some `compile_fail` tests that check that you _cannot_ derive `ShallowCopy` for types that contain non-`ShallowCopy` types.